File: Stack.cpp
#include "Stack.h"
#include <iostream>

void pushStack(NodeStack*& topStack, std::string value) {
    NodeStack* newNode = new NodeStack;
    newNode->key = value;
    newNode->next = topStack;
    topStack = newNode;
}

std::string popStack(NodeStack*& topStack) {
    if (topStack == nullptr) return "";
    NodeStack* temp = topStack;
    topStack = topStack->next;
    std::string value = topStack->key;
    delete temp;
    return value;
}

void printStack(NodeStack* topStack) {
    std::cout << "Stack: ";
    while (topStack != nullptr) {
        std::cout << topStack->key << " ";
        topStack = topStack->next;
    }
    std::cout << std::endl;
}

File: golang/stack.go
package main

import "fmt"

type NodeStack struct {
	key  string
	next *NodeStack
}

func pushStack(top **NodeStack, value string) {
	n := &NodeStack{key: value, next: *top}
	*top = n
}

func popStack(top **NodeStack) {
	if *top == nil {
		return
	}
	*top = (*top).next
}

func printStack(top *NodeStack) {
	p := top
	for p != nil {
		fmt.Print(p.key, " ")
		p = p.next
	}
	fmt.Println()
}
